### DISCUSSION TITLE
Proposal for a Collaborative Game Design Process

### DISCUSSION THREAD

**Discussion by jasper-one** [EveronLife/EveronLife#213] [Ideas]
**Introduction**
---
As Ark has indicated, our existing game design process presents bottlenecks.

I'm advocating for the integration of a roadmap-scoping process into our existing contribution process. This would empower everyone to contribute to the roadmap and more evenly distribute the game design work among our capable community.

**Proposed Solution**
---
I propose a revised process that utilizes both GitHub and Figma. The intention is to streamline our workflow, foster greater community engagement, and maintain the high-quality standards of Everon Life. 

Figma will serve as a centralized roadmap, enabling contributors to easily make updates and offering community members a straightforward way to understand ongoing and future development.

**Illustration with Example**
---
To demonstrate how this proposed process would work in practice, let's use an example: introducing a "Chop Shop" feature. The journey of this feature, from initial proposal to final implementation, will navigate through various stages of our design process. 

The **first two steps** of the process will remain the same as outlined in the [Contributing Guide](https://github.com/EveronLife/EveronLife/blob/main/.github/CONTRIBUTING.md)

**Newly Proposed Steps**

| Step | Description | Platform | GitHub Tags |
| ---- | ----------- | -------- | ----------- |
| 3. Go/No-Go | The core team reviews the feature request alongside other pending issues. If the team agrees that the feature is in alignment with the project's objectives and presents value, it's accepted for the roadmap. | GitHub Issues, Core Team Meetings | `planned` |
| 4. Add to Roadmap | The "Chop Shop" feature is incorporated into the roadmap on Figma, complete with a brief description, projected timeline, and assigned contributor(s), if applicable. | Figma | N/A |

The steps from **3 to 9 of the current process** will proceed as detailed in the [Contributing Guide](https://github.com/EveronLife/EveronLife/blob/main/.github/CONTRIBUTING.md)

**Final New Element**

| Step | Description | Platform | GitHub Tags |
| ---- | ----------- | -------- | ----------- |
| 10. Completion | Upon completion and merging of all tasks associated with the "Chop Shop" feature, the feature is marked as complete on the roadmap. The community is updated on the new feature, which will be included in the next mod update. | Figma, Discord, GitHub Issues | Issue Closed |

**Next Steps**
---
If we agree. I suggest that we document this in our contributor guide, notify the community, and define a few basic roles and responsibilities for team members to set things in motion. 

The end goal here is to deliver an first version of the roadmap that features documented discussions, issues, and is visually communicated via Figma.

**Comment by dan-schwerner** [comment]
How about using trello's github integration for better task tracking? with Figma we'd have to update the status of commits manually and would get out of date pretty quick + history tracking is a bit meh AND Adobe... https://blog.trello.com/github-and-trello-integrate-your-commits. I set up quite a few jira boards, can't imagine this to be any trickier. Having a bit of Kanban would definitely be nice. Just my two cents!

**Comment by MarvinVomberg** [comment]
I think the core team/maintainer are no source of truth if a feature should be included or not. Maintainer exist to keep the code working and orchestrating the work, while creating "generics" or features that are necessary to work.

Single Core Members are of course free to contribute to any "community features". Feature requests should be created with an understandable concept behind the feature, at best they include a technical assessment on how to implement it without creating code so far.

The Community should then decide by voting and discussing if a feature is necessary, the core team should only "veto" against game breaking changes.

So your third step should be like a poll if a feature is wanted by the community or not. Planning on when it is included and assessment on deprecation etc. can then still be done.

**Thats just my opinion though, I myself want to design the framework as open and collaborative as it can be without having a broken/bas performing end result**
